apps/expo/src/app/signincode.tsx

```diff
@@ -24,16 +28,35 @@ const SignInScreen = () => {
       return;
     }
 
+    const isDevModeUsingTestEmail = __DEV__ && code === "";
+
     try {
       const completeSignIn = await signIn.attemptFirstFactor({
         strategy: "email_code",
         code:
-          __DEV__ && code === "" && process.env.EXPO_PUBLIC_CLERK_TEST_CODE
+          isDevModeUsingTestEmail && process.env.EXPO_PUBLIC_CLERK_TEST_CODE
             ? process.env.EXPO_PUBLIC_CLERK_TEST_CODE
             : code,
       });
-      await setActive({ session: completeSignIn.createdSessionId });
-      router.replace("/account");
+
+      // If we have the id, that means the user successfully
+      // authenticated using Clerk. However we shouldn't update the
+      // active session just yet in case they're not approved
+      if (completeSignIn.id) {
+        const isApproved = await createUserWithClerkIdIfNotExists.mutateAsync(
+          completeSignIn.id,
```

Comment:
This actually isn't the user id, I'll have to change up the implementaiton a bit